bin/changelog

```diff
@@ -0,0 +1,303 @@
+
+  const issueLinkFragment =
+    `[#${issue}](https://github.com/handsontable/handsontable/issues/${issue})`;
+
+  return `${frameworkFragment}${title} ${issueLinkFragment}`;
+};
+
+const extractNumberFromString = (s) => {
+  const result = /\d+/.exec(s);
+
+  if (result === null) {
+    return undefined;
+  }
+
+  return Number(result[0]);
+};
+
+const askToProceedOrExit = async(message = 'Proceed?') => {
+  const { shouldProceed } = isTTY ? await inquirer.prompt({
+    name: 'shouldProceed',
+    type: 'confirm',
+    message,
+    default: true
+  }) : true;
+
+  if (shouldProceed === false) {
+    process.exit(1);
+  }
+};
+
+/**
+ * Checks if the changelog entry has all the necessary properties. Returns a
+ * new object with only the changelog entry properties that can be safely
+ * stringified and put into a file. If the input object is missing properties
+ * or they're in invalid format, throws an error.
+ *
+ * @param {object} input A valid changelog entry object.
+ * @returns {object} The `input` object but only with valid changelog entry properties.
+ */
+const assertChangelogEntryFormat = (input) => {
+  if (typeof input !== 'object' || input === null) {
+    throw new TypeError(`input must be an object and not be null (got: ${input})`);
+  }
+
+  if (typeof input.title !== 'string' || input.title.length === 0) {
+    throw new TypeError(`input.title must be a non-empty string (got: ${input.title})`);
+  }
+
+  if (typeof input.type !== 'string' || !changelogEntryTypes.includes(input.type)) {
+    throw new TypeError(`input.type must be one of: ${changelogEntryTypes} (got: ${input.type})`);
+  }
+
+  if (typeof input.issue !== 'number' || Number.isNaN(input.issue)) {
+    throw new TypeError(`input.issue must be a non-NaN number (got: ${input.issue})`);
+  }
+
+  if (typeof input.framework !== 'string' || !changelogFrameworkTypes.includes(input.framework)) {
+    throw new TypeError(`input.framework must be one of: ${changelogFrameworkTypes} (got: ${input.framework})`);
+  }
+
+  return {
+    title: input.title,
+    type: input.type,
+    issue: input.issue,
+    framework: input.framework
+  };
+};
+
+const createEntryCommand = async(args) => {
+  // https://stackoverflow.com/a/11958481
+  const gitBranchName = (await execa('git', ['rev-parse', '--symbolic-full-name', '--abbrev-ref', 'HEAD'])).stdout;
+  const maybeExtractedNumber = extractNumberFromString(gitBranchName);
+
+  const answers = isTTY ? await inquirer.prompt([
+    {
+      name: 'title',
+      type: 'input',
+      message: 'Title of the entry (e.g. "Fixed an issue with...")',
+      validate: s => s.length > 0,
+      when: typeof args.title === 'undefined'
+    },
+    {
+      name: 'issue',
+      type: 'input',
+      message: '# of the related issue/PR (e.g. 512)',
+      default: maybeExtractedNumber,
+      validate: s => ((Number.isNaN(Number(s)) || s.length === 0) ? 'The provided input is not a number' : true),
```

Comment:
I'd change the validator to accept only decimal numbers. Right now it is possible to explicitly pass the value in different number base systems (e.q `0xfff`, `0o10`) or even with scientific notation (e.q `4e4`). This can cause creating a file entry for other issue.

Reply:
Good catch, though I would argue that it would be pretty easy to spot if you made a mistake, even just during the prompting process anyway, as the value will get updated automatically:

https://user-images.githubusercontent.com/7091471/103890042-b1778680-50e7-11eb-94df-aa1b18f9df88.mov

Good suggestion, but imo it's too much, as you can easily mess up the issue number in other ways anyway.

Reply:
With `parseInt` I think it looks much more predictable. There is no magic conversation. Moreover, I've spotted that `filter` option returns `NaN` in the prompt for non-number values. That causes an issue where you are not able anymore to provide a valid value.

https://user-images.githubusercontent.com/571316/103894761-aaed0d00-50ef-11eb-926e-254fcca651a1.mov

I can able to fix the above by something like:
```
      validate: s => (Number.isInteger(s) ? true : 'The provided input is not a number'),
      filter: s => Number.isNaN(parseInt(s, 10)) ? '' : parseInt(s, 10),
```

Reply:
Sure, good enough for me https://github.com/handsontable/handsontable/pull/7448/commits/8b3808368329c34645a1a3be7393ad629fe4c5cd